src/core/public/http/fetch.ts

```diff
@@ -160,11 +171,14 @@ export class Fetch {
       throw new HttpFetchError(response.statusText, request, response, body);
     }
 
-    return new HttpResponse({ request, response, body });
+    return { fetchOptions, request, response, body };
   }
 
-  private shorthand(method: string) {
-    return (path: string, options: HttpFetchOptions = {}) =>
-      this.fetch(path, { ...options, method });
+  private shorthand(method: string): HttpHandler {
+    return (pathOrOptions: string | HttpFetchOptionsWithPath, options: HttpFetchOptions = {}) => {
+      const optionsWithPath: HttpFetchOptionsWithPath =
+        typeof pathOrOptions === 'string' ? { ...options, path: pathOrOptions } : pathOrOptions;
```

Comment:
note: it's a bit confusing that exposing `fetch` and shorthands we have to duplicate this logic.

Reply:
I agree that the signature is a little confusing. However the dual signature allow to answer the most common use case with shorter code (`.get(url)` vs `.get({url: url})`), and I'm not sure to see what exact kind of bug can appear, as long as we explicitly state that the first `path` parameter is prevalent on `options.path`. So I think that's a low-risk quick-win on developer experience, but I'm fine either way.

For the record, the Axios API:

```
request<T = any>(config: AxiosRequestConfig): AxiosPromise<T>;
get<T = any>(url: string, config?: AxiosRequestConfig): AxiosPromise<T>;
```

with `AxiosRequestConfig` containing also the `url` parameter, which is just ignored when calling with shorthands.

Also, as said on another comment, I would remove the distinction between `HttpFetchOptions` and `HttpFetchOptionsWithPath` and just document that when invoking with shorthands, `options.path` is ignored as the first parameter is prevalent.

Reply:
> and I'm not sure to see what exact kind of bug can appear, as long as we explicitly state that the first path parameter is prevalent on options.path

collision of any `HttpFetchOptions`, for example:
```js
.get({ headers: {a: 'a'}, path: '/'}, { headers: {b: 'b'} }
```

I'd suggest to overload signature to make it similar to axios:
```ts
fetch(options: HttpFetchOptions & {path: string})
fetch(path: string, options: HttpFetchOptions)
```

Reply:
Hum, good point,  I did not see at first look that `(pathOrOptions: string | HttpFetchOptionsWithPath, options: HttpFetchOptions = {})` typing allowed to pass options as first and second parameter. The overloaded signature makes sense then. 

`HttpFetchOptionsWithPath` is still used internally however, so I'm not a big fan of using `HttpFetchOptions & {path: string}` instead, as it will be defined at multiple places, but this is still fine I guess.

I would personally probably stick to adding `path?` to `HttpFetchOptions` to keep a single type,  as it seems the most pragmatic to me, but it will force to add validation and `!` access at some places, so I can understand not going that way.

Reply:
> I'd suggest to overload signature to make it similar to axios:
> 
> ```ts
> fetch(options: HttpFetchOptions & {path: string})
> fetch(path: string, options: HttpFetchOptions)
> ```

I'm not sure I understand how this is functionally different than what I already have. It leads to needing to include `& { path: string; }` in quite a few places, though it still has the potential collision bug you mentioned before (though I don't think that bug exists in the current implementation).

Reply:
Discussed offline. The confusion was from the `HttpHandler` return type not being super obvious. So from a TS perspective, this is good, but I will see if there's some better runtime protection for JS consumers we can add.